### DISCUSSION TITLE
Help needed : contributing to Sytem.Private.CoreLib

### DISCUSSION THREAD

**Discussion by CarlVerret** [dotnet/runtime#62000] [Q&A] [answered]
Hi there,
Following an issue that had been accepted by System.Runtime area manager, I'm trying to contribute to S.P.C. with a PR.
For some reason, I can't figure out how to add a new stuff to the double class.

Lets say I'd like to add a simple function like  this one (this is a simple example for sake of simplicity ) : 
```
public static bool ReturnFalse()=> false;
```
Been building S.P.C. with success and even after using  : 
`build.cmd clr.corelib+clr.nativecorelib+libs.pretest`
I still can't see this new function inside System.Runtime.Tests.  Having a hard time trying to understand the build worklfow....

Can someone please help me figure out  what I am missing ?  I am using VS2022 on Windows 10.

**Comment by huoyaoyuan** [comment, marked as answer]
A new public member must be defined in both *implementation* and *reference api*.
The reference api of System.Private.CoreLib should be in System.Runtime/ref.
